lib/src/collect_lcov.dart

```diff
@@ -104,26 +105,23 @@ class LcovCollector {
   /// Generates and returns a coverage json file
   Future<CoverageResult<List<File>>> _getCoverageJson(
       String testFile, Directory coverageDir) async {
+    int observatoryPort = 8181; // FIXME: Should read out port instead?
     var args = [
-      "--coverage_dir=${coverageDir.path}",
-      "--package-root=${packageRoot}",
+      "--packages=${packagesPath}",
+      "--pause-isolates-on-exit",
+      "--enable-vm-service=${observatoryPort}",
       testFile
     ];
-    var result = await processSystem.runProcess(Platform.executable, args);
-    if (result.exitCode < 0) {
-      stderr.writeln('stdout:');
-      stderr.writeln(result.stdout);
-      stderr.writeln('stderr:');
-      stderr.writeln(result.stderr);
-      throw new ProcessException(
-          Platform.executable,
-          args,
-          'There was a critical error. Exit code: ${result.exitCode}',
-          result.exitCode);
-    }
+    Future<ProcessResult> testFuture =
+        processSystem.runProcess(Platform.executable, args);
+    final Map data = await collect('localhost', observatoryPort, true, false);
```

Comment:
This will fail from SDK 1.22.0 on because observatory URI's will come with an access token for security reasons.
While [run_and_collect](https://github.com/dart-lang/coverage/blob/master/lib/src/run_and_collect.dart#L38) in `coverage` shows a solution - I would suggest to copy that behavior. 

Btw, I would advise against invoking `run_and_collect`directly because it only supports `--package-root` and also only starts the process, but never ends it, in my case resulting in observatory sometimes not starting.